### ISSUE TITLE
Loss and accuracy are always zero

### ISSUE THREAD

**Issue by mahmoudyusof** [google/trax#757]
### Description
Training doesn't affect loss nor accuracy which are always zero, I've tried building a model with the `Serial` Object and tried MLP
...

### Environment information
COLAB
```

$ pip freeze | grep tensor
mesh-tensorflow==0.1.13
tensor2tensor==1.15.7
tensorboard==2.2.2
tensorboard-plugin-wit==1.6.0.post3
tensorboardcolab==0.0.22
tensorflow==2.2.0
tensorflow-addons==0.8.3
tensorflow-datasets==2.1.0
tensorflow-estimator==2.2.0
tensorflow-gan==2.0.0
tensorflow-gcs-config==2.1.8
tensorflow-hub==0.8.0
tensorflow-metadata==0.22.2
tensorflow-privacy==0.2.2
tensorflow-probability==0.7.0

$ pip freeze | grep jax
jax==0.1.69
jaxlib==0.1.47

$ python -V
Python 3.6.9
```

#### Creating dataset `y = 2 * x - 1`
```python
def get_data(batch_size=16):
    xs = np.random.randint(low=1, high=10*batch_size, size=(batch_size, 1))
    ys = xs * 2 - 1;
    yield (xs, ys)

inputs = trax.supervised.Inputs(lambda _: get_data(16))
```

#### model and trainer
```python
def get_model(mode):
    # model = trax.layers.Serial(
    #     trax.layers.Dense(1)
    # )

    model = trax.models.PureMLP(layer_widths=(1, 1), out_activation=False, mode=mode)
    return model

output_dir = os.path.expanduser('~/train_dir/')
trainer = trax.supervised.Trainer(
    model=get_model,
    loss_fn=trax.layers.L2Loss(weight=1.0, batch_axis=0),
    optimizer=trax.optimizers.Adam,  # Change optimizer params here.
    lr_schedule=trax.lr.MultifactorSchedule,  # Change lr schedule here.
    inputs=inputs,
    output_dir=output_dir)

n_epochs  = 3
train_steps = 5
eval_steps = 2
for _ in range(n_epochs):
  trainer.train_epoch(train_steps, eval_steps)
```

The result is the following repeated three times
```
Step     20: Ran 5 train steps in 0.54 secs
Step     20: Evaluation
Step     20: train                   accuracy |  0.00000000
Step     20: train                       loss |  0.00000000
Step     20: train         neg_log_perplexity |  0.00000000
Step     20: train          sequence_accuracy |  0.00000000
Step     20: train weights_per_batch_per_core |  16.00000000
Step     20: eval                    accuracy |  0.00000000
Step     20: eval                        loss |  0.00000000
Step     20: eval          neg_log_perplexity |  0.00000000
Step     20: eval           sequence_accuracy |  0.00000000
Step     20: eval  weights_per_batch_per_core |  16.00000000
Step     20: Finished evaluation
```

**Comment by lukaszkaiser** [comment]
Can you also add the metrics parameter to Trainer, so metrics={'l2_loss': tl.L2Loss(...)} -- so it's the same as loss?

I think the problem is just metrics reporting: the default ones are for cross-entropy loss and they are misreporting L2, I think. But it'd be good to check.

**Comment by mahmoudyusof** [comment]
That did work, however wouldn't it make more sense for the trainer to infer which loss to report automatically ? I am willing to work on this
